slsDetectorGui/src/qTabDataOutput.cpp

```diff
@@ -416,6 +438,57 @@ void qTabDataOutput::SetParallel(bool enable) {
                  &qTabDataOutput::GetParallel)
 }
 
+void qTabDataOutput::GetCounterMask() {
+    LOG(logDEBUG) << "Getting counter mask";
+    disconnect(chkCounter1, SIGNAL(toggled(bool)), this,
+               SLOT(SetCounterMask()));
+    disconnect(chkCounter2, SIGNAL(toggled(bool)), this,
+               SLOT(SetCounterMask()));
+    disconnect(chkCounter3, SIGNAL(toggled(bool)), this,
+               SLOT(SetCounterMask()));
+    try {
+        auto retval = sls::getSetBits(det->getCounterMask().tsquash(
+            "Counter mask is inconsistent for all detectors."));
+        std::vector<QCheckBox *> counters = {chkCounter1, chkCounter2,
+                                             chkCounter3};
+        // default to unchecked
+        for (unsigned int i = 0; i < counters.size(); ++i) {
```

Comment:
Since you touch all elements maybe std::for_each is more clear?

Reply:
used range based for loop as you suggested. thanks!